scripts/check-rng-hygiene.sh

```
@@ -0,0 +1,145 @@
+#
+# What this does NOT cover, stated so nobody reads a green run as more than it
+# is: it is a grep. It catches the shape, not the intent. It cannot tell whether
+# a checked draw is used correctly once generated, it does not see into
+# managed_components/ or the sibling component repos (libnostr-c, noscrypt,
+# secp256k1-frost, libwally-core), and it cannot prove the SAR ADC is physically
+# producing noise on real silicon. Those stay review and bench questions.
+#
+# Run from anywhere. Exits non-zero with the offending lines.
+
+set -uo pipefail
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)